studio/components/interfaces/Settings/Integrations/IntegrationsSettings.tsx

```diff
@@ -0,0 +1,830 @@
+                          <div
+                            key={connection.id}
+                            className="relative flex flex-col -gap-[1px] [&>li]:pb-0"
+                          >
+                            <IntegrationConnectionItem
+                              showNode={false}
+                              key={connection.id}
+                              connection={connection}
+                              type={'GitHub' as IntegrationName}
+                              onDeleteConnection={onDeleteGitHubConnection}
+                              className="!rounded-b-none !mb-0"
+                            />
+
+                            <div className="border-b border-l border-r rounded-b-lg">
+                              <GitHubIntegrationConnectionForm
+                                connection={connection}
+                                integration={integration}
+                              />
+                            </div>
+                          </div>
+                        ))}
+                      </ul>
+                    </>
+                  ) : (
+                    <IntegrationConnectionHeader
+                      markdown={`### ${integration.connections.length} project ${pluralize(
+                        integration.connections.length,
+                        'connection'
+                      )} Repository connections for GitHub`}
+                    />
+                  )}
+                </div>
+              )
+            })}
+        </ScaffoldSectionContent>
+      </ScaffoldSection>
+    </ScaffoldContainer>
+  )
+
+  const GitHubIntegrationConnectionForm = ({
+    connection,
+    integration,
+  }: {
+    connection: IntegrationProjectConnection
+    integration: Integration
+  }) => {
+    const [open, setOpen] = useState(false)
+    const comboBoxRef = useRef<HTMLButtonElement>(null)
+    const projectContext = useProjectContext()
+
+    const githubProjectIntegration = integration?.connections.find(
+      (connection) => connection.supabase_project_ref === projectContext.project?.ref
+    )
+
+    const [repoOwner, repoName] = githubProjectIntegration?.metadata.name.split('/') ?? []
+
+    const { ui } = useStore()
+
+    const {
+      data: githubBranches,
+      error: githubBranchesError,
+      isLoading: isLoadingBranches,
+      isSuccess: isSuccessBranches,
+    } = useGithubBranchesQuery({
+      organizationIntegrationId: integration?.id,
+      repoOwner,
+      repoName,
+    })
+
+    const { mutate: updateBranch, isLoading: isUpdatingProdBranch } = useBranchUpdateMutation({
+      onSuccess: (data) => {
+        ui.setNotification({
+          category: 'success',
+          message: `Changed Production Branch to ${data.git_branch}`,
+        })
+        setOpen(false)
+      },
+    })
+
+    const { data: previewBranches, isLoading: isLoadingPreviewBranches } = useBranchesQuery(
```

Comment:
im guessing based on the way previewBranches here is used we dont need a UI state to handle isLoadingPreviewBranches, can remove - not used atm too